lib/rsk-utils.js

```diff
@@ -400,6 +398,19 @@ const findEventInBlock = async (rskTxHelper, eventName, fromBlockHashOrBlockNumb
   return null;
 };
 
+const findBridgeTransactionsInThisBlock = async (web3Client, blockHashOrBlockNumber) => {
+  const isLovell700AlreadyActive = await Runners.common.forks.lovell700.isAlreadyActive()
+
+  if (isLovell700AlreadyActive){
+    const bridgeTransactionParser = new BridgeTransactionParser(web3Client);
+    return bridgeTransactionParser.getBridgeTransactionsInThisBlock(blockHashOrBlockNumber);
+  }
+
+  const isFingerroot500AlreadyActive = await Runners.common.forks.fingerroot500.isAlreadyActive();
+  const bridgeTxParser = isFingerroot500AlreadyActive ? fingerrootBridgeTxParser : hopBridgeTxParser;
+  return bridgeTxParser.getBridgeTransactionsInThisBlock(web3Client, blockHashOrBlockNumber);
```

Comment:
following some discussions we had about early return, we could refactor this to this:

```
  const isFingerroot500AlreadyActive = await Runners.common.forks.fingerroot500.isAlreadyActive();
  if (!isFingerroot500AlreadyActive) {
    return hopBridgeTxParser.getBridgeTransactionsInThisBlock(web3Client, blockHashOrBlockNumber);
  }

  const isLovell700AlreadyActive = await Runners.common.forks.lovell700.isAlreadyActive()
  if (!isLovell700AlreadyActive){
    return fingerrootBridgeTxParser.getBridgeTransactionsInThisBlock(web3Client, blockHashOrBlockNumber);
  }

  const bridgeTransactionParser = new BridgeTransactionParser(web3Client);
  return bridgeTransactionParser.getBridgeTransactionsInThisBlock(blockHashOrBlockNumber);
```

Reply:
I'd leave it as is, some day we'll get rid of the regression tests here and many of these checks won't be necessary anymore